.circleci/config.yml

```diff
@@ -28,6 +28,10 @@ references:
     - "1.21.4"
     - "1.22.2"
     - "1.23.0"
+  nomad-versions: &supported_nomad_versions
+    - &default_nomad_version "1.3.3"
+    - "1.2.10"
+    - "1.1.16"
```

Comment:
Could add a nightly job that runs `https://api.releases.hashicorp.com/v1/releases/nomad?limit=1` to detect new versions?

Reply:
I was thinking of it in a similar way to how we manage the Envoy versions, only simpler since there are fewer tests. I do like the idea of having some automation in place to keep this up to date though. Would the nightly job create a PR whenever it finds new version? I suppose if that too was pretty simple, it would be worth the toil prevention of having to keep this up-to-date ourselves. 

In any case, I think it's definitely worth automating this part too, but I'd do it in a separate PR.